chain/client/src/sync.rs

```diff
@@ -672,6 +672,21 @@ impl StateSync {
         Ok((update_sync_status, all_done))
     }
 
+    fn get_epoch_start_sync_hash(
+        &self,
+        chain: &mut Chain,
+        sync_hash: CryptoHash,
+    ) -> Result<CryptoHash, near_chain::Error> {
+        let mut hash = sync_hash;
+        loop {
+            let header = chain.get_block_header(&hash)?;
+            if header.inner_lite.next_epoch_id.0 == header.prev_hash {
```

Comment:
In general we should avoid compare epoch id with block hash. Use methods exposed in runtime adapter instead.

Reply:
@bowenwang1996 could you please point which exact methods are better to use instead?

Reply:
Given that you are already in a loop, just comparing epoch id to the prev epoch id should be sufficient?

Reply:
++